R/dtree_cross_validation.R

```diff
@@ -0,0 +1,531 @@
+    outVec <- c(mid = modelIndic, trial = trialIndic, fold = foldIndic, Accuracy_Overall = overallAcc, Accuracy_Class_1 = percentClass1Right, Accuracy_Class_2 = percentClass2Right, F1 = F1, AUC = AUC)
+  } else {
+    #Compute accuracy by class
+    outVec <- vector(length = length((extras$levels)))
+    for (l in 1:length((extras$levels))) {
+      tempPred <- scoredOutput[actual == (extras$levels)[[l]]]
+      nCorrect <- sum(tempPred == (extras$levels)[[l]])
+      thisAcc <- nCorrect/sum(actual == (extras$levels)[[l]])
+      outVec[l] <- thisAcc
+      names(outVec)[l] <- paste0("Accuracy_Class_", l)
+    }
+    outVec <- c(mid = modelIndic, trial = trialIndic, fold = foldIndic, Accuracy_Overall = overallAcc, outVec)
+  }
+  return(outVec)
+}
+
+# Functions to Generate Output
+#' @import reshape2
+generateConfusionMatrices <- function(outData, extras) {
+  outvec <- vector(length = length(extras$levels))
+  pasteClass <- function(nameOfClass) {
+    paste0("Class_", nameOfClass)
+  }
+  names(outvec) <- sapply(X = (extras$levels), FUN = pasteClass, simplify = TRUE)
+  for (i in 1:length(extras$levels)) {
+    outvec[i] <- length(which((outData$actual) == ((extras$levels)[i])))
+  }
+  return(c(mid = unique(outData$mid), trial = unique(outData$trial), fold = unique(outData$fold), Predicted_class = as.character(unique(outData$response)), outvec))
+}
+
+generateOutput3 <- function(data, extras, modelNames) {
+  d <- plyr::ddply(data, c("trial", "fold", "mid", "response"), generateConfusionMatrices,
+             extras = extras
+  )
+  d$Model <- modelNames[as.numeric(d$mid)]
+  d$Type <- rep.int('Classification', times = length(d$Model))
+  d <- subset(d, select = -c(mid, response))
+  d <- reshape2::melt(d, id = c('trial', 'fold', 'Model', 'Type', 'Predicted_class'))
+  colnames(d) <- c('Trial', 'Fold', 'Model', 'Type', 'Predicted_class', 'Variable', 'Value')
+  return(d)
+}
+
+generateOutput2 <- function(data, extras, modelNames) {
+  fun <- if (is.null(extras$levels)) {
+    getMeasuresRegression
+  } else {
+    getMeasuresClassification
+  }
+  d <- plyr::ddply(data, c("trial", "fold", "mid"), fun, extras = extras)
+  d$Model <- modelNames[as.numeric(d$mid)]
+  d <- subset(d, select = -c(mid))
+  return(d)
+}
+
+#' @import plyr
+#' @import rpart
+#' @import C50
+generateOutput1 <- function(inputs, config, extras){
+  allFolds <- extras$allFolds
+  g <- expand.grid(
+    mid = seq_along(inputs$models),
+    trial = seq_along(allFolds),
+    fold = seq_along(allFolds[[1]])
+  )
+  return(mdply(g, getCrossValidatedResults(inputs, allFolds, extras, config)))
+}
+
+
+#' Get the necessary measures in the binary classification case
+#'
+#' @param pred_prob vector of predicted probabilities
+#' @param actual vector of actual results
+#' @param threshold a double between 0 and 1 (current probability threshold)
+#' @return a data.frame with results
+#' @import ROCR
```

Comment:
Use `importFrom` to prevent namespace clobbering and warnings. This will get rid of all the warnings associated with ROCR. Also some place appropriate, you need to add

```r
#' @importFrom stats cor update
```

This is a new R 3.3.x requirement, to include namespace imports from recommended packages as well.